### PR TITLE
Update for AOS 4.0.0

### PR DESCRIPTION
**Pull request by huttmf**
Includes support for new probe message types and probe label.

EvpnType3RouteEvent
EvpnType5RouteEvent
InterfaceCountersUtilization
SystemInterfaceUtilization
ActiveFloodlistEvent

### CODE REVIEW

File: plugins/inputs/aos/aos_test.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ func TestExtractProbeMessage(t *testing.T) {
 	plugin := &Aos{
 		Port:            PORT,
 		Address:         "127.0.0.1",
-		StreamingType:   []string{"alerts"},
+		StreamingType:   []string{"perfmon"},
 		AosServer:       "127.0.0.1",
```
Comment (rmahidhar) [inline plugins/inputs/aos/aos_test.go:29]: streaming type changed in 4.0.0?

File: plugins/inputs/aos/aos_test.go [hunk 2]
```diff
@@ -44,17 +44,34 @@ func TestExtractProbeMessage(t *testing.T) {
...
 	tags["device"] = "probe_msg_1"
+	tags["probe_label"] = "probe_msg_label"
+
+	state := aos_streaming.RouteState_ROUTE_ADD
+	system_id := "1234567"
+	vni := uint32(5)
+	next_hop := "10.1.1.1"
+	rd := "100"
+	rt := "100"
+	evpn_type_3_route_event := &aos_streaming.EvpnType3RouteEvent{
```
Comment (rmahidhar) [inline plugins/inputs/aos/aos_test.go]: don't need to add test for evpn_type_5 route event?

File: plugins/inputs/aos/aos_test.go (hunk 1 above)
Comment (huttmf) [reply plugins/inputs/aos/aos_test.go:29]: No - even in 3.2.0 the probe_message was part of PerfMon so I changed it to reflect the actual message type.
```
message PerfMon {
  oneof data {
     InterfaceCounters interface_counters = 1;
     SysResourceCounters system_resource_counters = 2;
     GenericPerfmonMessage generic = 4;
     ProbeMessage probe_message = 5;
  }
  optional float time_delta = 3;
}
```

File: plugins/inputs/aos/aos_test.go (hunk 2 above)
Comment (huttmf) [reply plugins/inputs/aos/aos_test.go]: This test really only checks the tags. We do have other tests that actually check the fields using `	acc.AssertContainsTaggedFields(t, "event_streaming", fields, tags)`

I can look at adding a test like this for probe message.

Comment (huttmf) [reply plugins/inputs/aos/aos_test.go]: I added a more comprehensive test for EvpnType3RouteEvent for probe messages.

**Review by rmahidhar — APPROVED** [review]

File: plugins/inputs/aos/aos_streaming/aosstream.pb.go [hunk 3]
```diff
@@ -1,11 +1,11 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
-// source: aosstream.proto
+// source: aosstream-enduro.proto
```
Comment (breichert42) [inline plugins/inputs/aos/aos_streaming/aosstream.pb.go:2]: Is this manual or generated? Might be good to leave code names out.

**Review by breichert42 — APPROVED** [review]
New changes look good, if possible dropping Enduro would be good.

File: plugins/inputs/aos/aos_streaming/aosstream.pb.go (hunk 3 above)
Comment (huttmf) [reply plugins/inputs/aos/aos_streaming/aosstream.pb.go:2]: yeah it's auto generated, but since it's just a comment I'll take it out.

**Pull request merged**